packages/salesforce-adapter/src/filters/convert_maps.ts

```diff
@@ -99,21 +111,22 @@ const createOrderedMapType = <T extends TypeElement>(innerType: T): ObjectType =
     },
   })
 
-/**
- * Convert a string value into the map index keys.
- * Note: Reference expressions are not supported yet (the resolved value is not populated in fetch)
- * so this filter has to run before any filter adding references on the objects with the specified
- * metadata types (e.g Profile).
- */
-export const defaultMapper = (val: string): string[] => val.split(API_NAME_SEPARATOR).map(v => naclCase(v))
+const stringifyMapperInput = (mapperInput: MapperInput): string =>
+  _.isString(mapperInput) ? mapperInput : apiNameSync(mapperInput.value) ?? ''
+
+const isMapperInput = (val: unknown): val is MapperInput =>
+  _.isString(val) || isResolvedReferenceExpressionToElement(val)
+
+export const defaultMapper: MapperFunc = val =>
+  (_.isString(val) ? val : apiNameSync(val.value) ?? '').split(API_NAME_SEPARATOR).map(v => naclCase(v))
```

Comment:
```suggestion
  stringifyMapperInput(val).split(API_NAME_SEPARATOR).map(v => naclCase(v))
```
(I'm assuming this is what you meant to do)

Reply:
Did you miss this?